### PR TITLE
[autoscaler] Always restart cloudwatch agent to support existing installations.

### PR DESCRIPTION
**Pull request by pdames**
## Why are these changes needed?

This change resolves failures that otherwise occur when attempting to start the cloudwatch agent on a Ray cluster whenever the cloudwatch unified agent is already installed and running on 1 or more nodes but wasn't configured via our CloudwatchHelper class during `ray up`. For example, this failure is encountered when launching a new cluster from an existing Amazon Ray AMI in a new AWS account with the following autoscaler config section specified:
```yaml
provider:
    type: aws
    region: us-east-1
    availability_zone: us-east-1a, us-east-1b, us-east-1c, us-east-1d, us-east-1f
    cloudwatch:
      agent:
        config: "./python/ray/autoscaler/aws/cloudwatch/example-cloudwatch-agent-config.json"
```

To resolve this error, this change ensures that we never just try to start the cloudwatch agent, but first ensure that any existing cloudwatch agent instances are stopped (ignoring any failures that occur if it is not installed) before issuing a subsequent SSM command to start the cloudwatch agent on all nodes in the cluster.

Also some minor refactoring of prometheus/cloudwatch integration:
1. Example metric namespace renamed from `{cluster-name}-Prometheus` to `{cluster-name}-ray-prometheus`.
2. Always install any prometheus version in `/opt/prometheus` instead of `/opt/prometheus-{version}-{platform}` to reduce changes across files referencing the prometheus install directory over time.
3. Remove the prometheus installation package after installation completes.

## Checks

- [X] I'm working against the latest source on the **experimental** branch.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests.
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: python/ray/autoscaler/_private/aws/cloudwatch/cloudwatch_helper.py [hunk 1]
```diff
@@ -171,7 +172,7 @@ def ssm_install_cloudwatch_agent(self):
             "AWS-RunShellScript",
             parameters_run_shell,
         )
-        self._start_cloudwatch_agent(cwa_config_ssm_param_name)
+        self._restart_cloudwatch_agent(cwa_config_ssm_param_name)
```
Comment (Zyiqin-Miranda) [inline python/ray/autoscaler/_private/aws/cloudwatch/cloudwatch_helper.py:175]: Thanks Patrick for fixing the issue of starting cloudwatch agent when agent is already configured through AMI and running. All changes looks good to me. I’ve tested the scenario of stopping the agent with no agent running here and everything works.

**Comment by Zyiqin-Miranda** [comment]
Thanks Patrick for fixing the issue of starting cloudwatch agent when agent is already configured through AMI and running. All changes looks good to me.

**Review by Zyiqin-Miranda — APPROVED** [review]

File: python/ray/autoscaler/aws/cloudwatch/prometheus_ray_up.sh [hunk 2]
```diff
@@ -21,19 +21,19 @@ if [ $? -eq 0 ]; then
   # TODO remove the version number hardcoding from this script
   # download the latest prometheus
   ray exec "$AUTOSCALER_CONFIG_FILE" "sudo wget https://github.com/prometheus/prometheus/releases/download/v2.25.0/prometheus-2.25.0.linux-amd64.tar.gz -P /opt"
-  ray exec "$AUTOSCALER_CONFIG_FILE" "cd /opt && sudo tar xvfz prometheus-2.25.0.linux-amd64.tar.gz"
+  ray exec "$AUTOSCALER_CONFIG_FILE" "cd /opt && sudo tar xvfz prometheus-2.25.0.linux-amd64.tar.gz && sudo mv prometheus-2.25.0.linux-amd64 prometheus && sudo rm prometheus-2.25.0.linux-amd64.tar.gz"
```
Comment (annaluo676) [inline python/ray/autoscaler/aws/cloudwatch/prometheus_ray_up.sh:24]: Out of this PR's scope: shall we install the latest stable (ray-tested) version of Prometheus

**Review by annaluo676 — APPROVED** [review]

File: python/ray/autoscaler/aws/cloudwatch/prometheus_ray_up.sh (hunk 2 above)
Comment (pdames) [reply python/ray/autoscaler/aws/cloudwatch/prometheus_ray_up.sh:24]: Based on @Zyiqin-Miranda's recent test results, it looks like we don't need to install Prometheus at all to scrape Ray system metrics, so I'll remove this in a subsequent revision.

**Pull request merged**